FlowCrypt/Controllers/Attachment/AttachmentViewController.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  AttachmentViewController.swift
+//  FlowCrypt
+//
+//  Created by Evgenii Kyivskyi on 11/16/21
+//  Copyright © 2017-present FlowCrypt a. s. All rights reserved.
+//
+
+import UIKit
+import WebKit
+import FlowCryptUI
+import Combine
+
+@MainActor
+final class AttachmentViewController: UIViewController {
+
+    private lazy var webView = WKWebView()
+    private let file: FileType
+    private let shouldShowDownloadButton: Bool
+
+    private let filesManager: FilesManagerType
+    private lazy var attachmentManager = AttachmentManager(
+        controller: self,
+        filesManager: filesManager
+    )
+
+    private let errorLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textAlignment = .center
+        label.text = "no_preview_avalable".localized
+        label.isHidden = true
+        label.textColor = .textColor
+        label.font = .systemFont(ofSize: 12)
+        return label
+    }()
+
+    init(
+        file: FileType,
+        shouldShowDownloadButton: Bool,
+        filesManager: FilesManagerType = FilesManager()
+    ) {
+        self.file = file
+        self.shouldShowDownloadButton = shouldShowDownloadButton
+        self.filesManager = filesManager
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupNavigationBar()
+        addWebView()
+        showSpinner()
+        title = file.name
+        saveAndStartDownload(file: file)
+    }
+
+    private func remove(file: FileType) {
+        Task { [weak self] in
+            do {
+                try await filesManager.remove(file: file)
+                self?.navigationController?.popViewController(animated: true)
+            } catch let error {
+                showAlert(
+                    message: "could_not_remove_opened_attachment".localized + " (\(error.localizedDescription))"
+                ) {
+                    self?.navigationController?.popViewController(animated: true)
+                }
+            }
+        }
+    }
+
+    private func saveAndStartDownload(file: FileType) {
+        Task { [weak self] in
+            do {
+                let url = try await filesManager.save(file: file)
+                self?.load(with: url)
+            } catch let error {
+                showAlert(
+                    message: "could_not_open_an_attachment".localized + " (\(error.localizedDescription))"
+                ) {
+                    self?.navigationController?.popViewController(animated: true)
+                }
+            }
+        }
+    }
```

Comment:
If I understand this correctly, here is what happens:
 - I tap on the attachment to preview it
 - attachment gets automatically saved somewhere (unsure where, not familiar with iOS environment)
 - attachment gets previewed automatically
 - attachment gets removed (hopefully, unless there's an error, and then it's left hanging)

I'd like to avoid saving it completely, and go straight from memory to preview without intermediate steps. You should be able to provide it to the webview as a blob directly. Is that possible?

Reply:
The WebView can be loaded with URLRequest only, that need to have the path. 
Maybe @sosnovsky has ever dealt with opening files in WebView from memory?

Reply:
What is the issue with temporary saving the file?

Reply:
Security - if you drop your phone and someone unlocks and roots it, they get your attachment content. They would not be able to get that if it was kept in memory.

Consistency - files are not cleared, as implemented, if errors happened after saving them but before deleting them.

Consistency 2 - files are not cleared if there is error while clearing them.

Consistency 3 - if app crashes in between opening file and removing it, files are left over.

These three consistency issues worsen the security problem - there is bound to be files left over after you've used the app a lot, and since you bothered to use encryption in the first place, they are likely very sensitive files.

Bettery - doing work you would not need to otherwise do consumes battery. This is minor compared to the security issue.

Complexity - if it was possible to directly view the file, it would surely be simpler to implement than saving and deleting files. (also minor compared to security)

Reply:
Here you can directly load from `Data` and a mime type - seems exactly what we need.

https://developer.apple.com/documentation/webkit/wkwebview/1415011-load

```swift
func load(
  _ data: Data, 
  mimeType MIMEType: String, 
  characterEncodingName: String, 
  baseURL: URL
) -> WKNavigation?
```

Reply:
Just found it and trying to handle that.